restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/AccountsEndpoint.java

```diff
@@ -597,6 +583,118 @@ protected Response updateAccount(final String identifier, final MultivaluedMap<S
         }
     }
 
+    private Account getOperatingAccount (String identifier) {
+        Sid sid = null;
+        Account account = null;
+        try {
+            sid = new Sid(identifier);
+            account = accountsDao.getAccount(sid);
+        } catch (Exception e) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Exception trying to get account using SID. Seems we have email as identifier");
+            }
+        }
+        if (account == null) {
+            account = accountsDao.getAccount(identifier);
+        }
+        return account;
+    }
+
+    private Organization getOrganization(final MultivaluedMap<String, String> data) {
+        Organization organization = null;
+        Sid organizationSid = null;
+        String organizationId = null;
+
+        if (data.containsKey("Organization")) {
+            organizationId = data.getFirst("Organization");
+        } else {
+            return null;
+        }
+
+        //Attempt to get Organization by domain name
+        organization = organizationsDao.getOrganizationByDomainName(organizationId);
+
+        if (organization != null) {
+            return organization;
+        } else {
+            //Attempt to match organizationId to organizationSid
+            try {
+                organizationSid = new Sid(organizationId);
+                organization = organizationsDao.getOrganization(organizationSid);
+                return organization;
+            } catch (IllegalArgumentException e) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Exception while tried to match organizationId to organizationSid");
+                }
+            }
+        }
+        return organization;
+    }
+
+    protected Response migrateAccountOrganization(final String identifier, final MultivaluedMap<String, String> data,
+                                              final MediaType responseType) {
+
+        //Validation 1 - Only SuperAdmin is allowed to migrate organization for an Account
+        if (!isSuperAdmin()) {
+            throw new InsufficientPermission();
+        }
+
+        Organization organization = getOrganization(data);
+        //Validation 2 - Check if data contains Organization (either SID or domain name)
+        if (organization == null) {
+            return status(PRECONDITION_FAILED).entity("Missing Organization SID or Domain Name").build();
+        }
+
+        Account operatingAccount = getOperatingAccount(identifier);
+
+        //Validation 3 - Operating Account shouldn't be null;
+        if (operatingAccount == null) {
```

Comment:
clarification:what is the meaning/purpose of this OperatingAccoutn check here?

Reply:
@jaimecasero this is a fail fast check, if the `identifier` passed in the request, doesn't match an existing account, then request should fail fast.